app/lib/features/spaces/widgets/space_list_widget.dart

```diff
@@ -0,0 +1,92 @@
+import 'package:acter/common/toolkit/errors/error_page.dart';
+import 'package:acter/common/widgets/room/room_card.dart';
+import 'package:acter/features/space/widgets/space_sections/section_header.dart';
+import 'package:acter/features/spaces/providers/space_list_provider.dart';
+import 'package:acter/features/spaces/widgets/space_list_skeleton.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:logging/logging.dart';
+
+final _log = Logger('a3::space-list-widget');
+
+class SpaceListWidget extends ConsumerWidget {
+  final String? searchValue;
+  final int? limit;
+  final bool showSectionHeader;
+  final VoidCallback? onClickSectionHeader;
+  final bool shrinkWrap;
+  final Widget emptyState;
+
+  const SpaceListWidget({
+    super.key,
+    this.limit,
+    this.searchValue,
+    this.showSectionHeader = false,
+    this.onClickSectionHeader,
+    this.shrinkWrap = true,
+    this.emptyState = const SizedBox.shrink(),
+  });
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final spaceLoader = ref.watch(spaceListSearchProvider(searchValue ?? ''));
+    return spaceLoader.when(
+      data: (spaceList) => buildSpaceSectionUI(context, spaceList),
+      error: (error, stack) => spaceListErrorWidget(context, ref, error, stack),
+      loading: () => const SpaceListSkeleton(),
+    );
+  }
+
+  Widget spaceListErrorWidget(
+    BuildContext context,
+    WidgetRef ref,
+    Object error,
+    StackTrace stack,
+  ) {
+    _log.severe('Failed to load spaces', error, stack);
+    return ErrorPage(
+      background: const SpaceListSkeleton(),
+      error: error,
+      stack: stack,
+      textBuilder: L10n.of(context).loadingFailed,
+      onRetryTap: () {
+        ref.invalidate(spaceListSearchProvider(searchValue ?? ''));
+      },
+    );
+  }
+
+  Widget buildSpaceSectionUI(BuildContext context, List<Space> spaceList) {
+    if (spaceList.isEmpty) return emptyState;
+
+    final count = (limit ?? spaceList.length).clamp(0, spaceList.length);
+    return showSectionHeader
+        ? Column(
+            mainAxisSize: MainAxisSize.min,
+            children: [
+              SectionHeader(
+                title: L10n.of(context).spaces,
+                isShowSeeAllButton: count < spaceList.length,
```

Comment:
as this PR doesn't contain any active version of `showSectionHeader` I am not sure what this does in here, but I _suspect_ this is meant for the future use case of using this as the embedded dashboard widget as well: if that is the case I want to point out that if we showed only the bookmarks, then the spaceList.length might as well be >= count yet we want to have the `isShwoSeeAllButton` visible regardless _if_ there are more spaces _in general_. In case of bookmarks we currently show _all_ bookmarks (as the order _within_ bookmarks isn't deterministic and thus the user can be very annoyed if they bookmarked a spaces but it doesn't show up) and only limit in the non-bookmarked case.

Just to keep in mind how this is meant to work ...

Reply:
Yes this PR doesn't contain `SectionHeader` work but I already started work on [Next Quick Search PR](https://github.com/acterglobal/a3/pull/2275) on top of this PR where it is require.

<img width="359" alt="Screenshot 2024-10-17 at 11 29 33 AM" src="https://github.com/user-attachments/assets/0661101c-de01-4058-a565-cbd9f14425a8">

I understand the use-case of my bookmarks section of dashboard and that's why haven't touch it at the moment with this code. But I needed section header stuff in Quick search PR as shown in above screenshot.